typescript/looker/rtl/nodeTransport.ts

```diff
@@ -394,20 +394,29 @@ export class NodeTransport implements ITransport {
 
 async function parseResponse(contentType: string, res: Response) {
   const mode = responseMode(contentType)
+  const utf8 = 'utf8'
   let result = await res.body
   if (mode === ResponseMode.string) {
-    if (!isUtf8(contentType)) {
-      // always convert to UTF-8 from whatever it was
-      result = Buffer.from(result.toString(), 'utf8')
-    }
     if (contentType.match(/^application\/.*\bjson\b/g)) {
       try {
-        result = result instanceof Object ? result : JSON.parse(result)
-        return result
+        if (result instanceof Buffer) {
+          result = result.toString(utf8)
+        }
+        if (result instanceof Object) {
+          return result
+        }
+        return JSON.parse(result.toString())
```

Comment:
I don't really follow this part. Why do we have to decode the buffer into a string using 'utf-8' encoding and then decode it again before passing it to `JSON.parse()`? I also looks like `toString()` defaults to 'utf-8' encoding.

Reply:
I'm still trying to reason through this part and will need more comprehensive testing. Multiple changes in 6.22+ broke `parseResponse` and this is what I got done as a band-aid for now. I will tag that routine with a `TODO` for more cleanup.